File: tests/fixtures/python/classes.py
"""Test fixture: Python classes with methods for symbol extraction testing."""

from dataclasses import dataclass


class Calculator:
    """A simple calculator class."""

    def __init__(self, initial_value: float = 0.0):
        """Initialize calculator with a value."""
        self.value = initial_value

    def add(self, x: float) -> float:
        """Add a number to the current value."""
        self.value += x
        return self.value

    def subtract(self, x: float) -> float:
        """Subtract a number from the current value."""
        self.value -= x
        return self.value

    def reset(self) -> None:
        """Reset the calculator to zero."""
        self.value = 0.0


@dataclass
class Point:
    """A 2D point."""

    x: float
    y: float

    def distance_from_origin(self) -> float:
        """Calculate distance from origin."""
        return (self.x**2 + self.y**2) ** 0.5


class _PrivateHelper:
    """A private helper class (convention)."""

    def _internal_method(self) -> None:
        """Internal method."""
        pass

File: tests/fixtures/typescript/classes.ts
/**
 * Test fixture: TypeScript classes with methods for symbol extraction testing.
 */

export class Calculator {
    private value: number;

    constructor(initialValue: number = 0) {
        this.value = initialValue;
    }

    public add(x: number): number {
        this.value += x;
        return this.value;
    }

    public subtract(x: number): number {
        this.value -= x;
        return this.value;
    }

    protected reset(): void {
        this.value = 0;
    }
}

interface Point {
    x: number;
    y: number;
}

class Point2D implements Point {
    constructor(public x: number, public y: number) {}

    distanceFromOrigin(): number {
        return Math.sqrt(this.x ** 2 + this.y ** 2);
    }
}

abstract class Shape {
    abstract area(): number;
}
